patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
```

Comment:
@vkurchatkin that's just how you're asked to write question/answer pairs in StackOverflow - I posted it there because there was an older (and frankly worse) article about it and I wanted people searching for it to find something meaningful first.

The name is just what I saw people call it before me - I'm not attached to it in particular. The pattern can also be called disposers (C#) , context managers (Python), auto-closables (Java), RAII (C++) and in many other names.